cpp/include/cugraph.h

```diff
@@ -25,4 +25,6 @@
 
 #include "functions.h"
 
+#include "algorithms.h"
```

Comment:
Why do we need `algorithms.h` to be separated from `functions.h`? 
This may lead to an inconsistency with cuDF which is using `functions.h` only.

Reply:
The reasoning is to avoid having a too large file with everything in it. functions.h covers functions on the graph object and algorithms covers algorithms which act on a graph object.

Reply:
Ok, let's keep `algorithms.h`